src/gui/models/procedureModel.cpp

```diff
@@ -102,6 +126,63 @@ QVariant ProcedureModel::data(const QModelIndex &index, int role) const
     return {};
 }
 
+bool ProcedureModel::setData(const QModelIndex &index, const QVariant &value, int role)
+{
+    if (!procedure_)
+        return false;
+
+    if (role == Qt::EditRole)
+    {
+        auto *node = static_cast<ProcedureNode *>(index.internalPointer());
```

Comment:
```suggestion
        auto *node = rawData(index);
```
I like your `rawData` function and it seems a shame not to use it.